### DISCUSSION TITLE
Defining steps with the useTimeField

### DISCUSSION THREAD

**Discussion by jussivirtanen** [adobe/react-spectrum#4400] [Ideas]
In our project, we'd like to make the `useTimeField` minute selector accept only specific increments (0, 15, 30 and 45). With the native `<input type="time">` element, it's possible to use the [step](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time#step) attribute to determine the granularity in seconds.

It would be great to have this functionality with the [useTimeField](https://react-spectrum.adobe.com/react-aria/useTimeField.html)! And, if there _is_ a way to get this functionality working with the existing `useTimeField` version, I'm thrilled to hear about that. Thanks!

**Comment by snowystinger** [comment]
I think we could consider adding this to make it easier.

I'm not sure we'd want to add it quite like the native control. It's a bit weird to use. The step value applies across every segment and controls if there is a seconds field or not. If you specify something like 150, it just pretends it was never specified. And it looks like even the spec has some outstanding issues https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time#step

**Reply by jussivirtanen** [reply]
Thanks for the quick reply, @snowystinger and great to hear that you could consider this. I agree that the native picker is quirky at best and behaves oddly with the step attribute. Looking forward to seeing if this gets pushed forward at some point!

**Reply by snowystinger** [reply]
I've opened an issue to cover it, linked here #4410

**Reply by jussivirtanen** [reply]
Super, thanks! 🙏

**Comment by crazko** [comment]
Please, have a look at https://github.com/adobe/react-spectrum/issues/4410#issuecomment-1521365424 where I proposed a solution.

Any improvements are welcome!
